Add troubleshooting section with exposing-ports sub-section to containers local-dev docs (#23495)
cloudflare/cloudflare-docs: src/content/docs/containers/local-dev.mdx

into the local container images to sync code changes. This can be done, but there is no built-in
mechanism for doing so in Wrangler, and best-practices will depend on the languages and frameworks
you are using in your container code.

## Troubleshooting

### Exposing Ports

In production, all of your container's ports will be accessible by your Worker,
so you do not need to specifically expose ports using the [`EXPOSE` instruction](https://docs.docker.com/reference/dockerfile/#expose) in your Dockerfile.

> But for local development on non-Linux systems, you most likely need to expose a port to access your container properly; for example: `EXPOSE 4000`.

Reviewer: ```suggestion
But for local development on non-Linux systems, you will need to declare any ports you need to access in your Dockerfile with the EXPOSE instruction; for example: `EXPOSE 4000`, if you will be accessing port 4000.
```

> If you have not exposed any ports, you will see the following type of error when building or pulling your container image:

Reviewer: ```suggestion
If you have not exposed any ports, you will see the following error in local development:
```

```text
The container "MyContainer" does not expose any ports, please expose any port that you want to use with `getTCPPort()`.
```

> And if you call `getTCPPort()` on an unexposed port you will see the following error:

Reviewer: ```suggestion
And if you call `getTCPPort()` on a port that you have not exposed in your `Dockerfile` you will see the following error:
```

```text

> XXX

Author: TODO

> ```

Reviewer: ```suggestion
```
You may also see this while the container is starting up and no ports are available yet. You should retry until the port becomes available. This retry logic should be handled for you if you are using the [containers package](https://github.com/cloudflare/containers/tree/main/src).
```
